kernel/cls/cls_macros/src/lib.rs

```diff
@@ -264,3 +273,221 @@ pub fn cpu_local(args: TokenStream, input: TokenStream) -> TokenStream {
     }
     .into()
 }
+
+fn cls_offset_expr(name: &Ident) -> proc_macro2::TokenStream {
+    quote! {
+        {
+            // There are two ways in which a crate can be linked:
```

Comment:
this documentation is really awesome, I think it should be promoted to a top crate-level doc so it gets rendered by rustdoc, which also allows us to link to it.

The same goes for the comment blocks starting at line 324 and 431.

Then, you can reference the crate-level docs with inline code comments here.